beacon/validation/fields.py

```diff
@@ -219,7 +219,9 @@ def __init__(self, *, items=None, separator=',', trim=True, **kwargs):
     async def convert(self, value: str, **kwargs) -> set:
         if value in EMPTY_VALUES:
             return self.default
-        if isinstance(value, list):
+        if value is None:
+            values = []
```

Comment:
The annotation said it should return a set. Not the case here.

Reply:
What do you mean?

```py
async def convert(self, value: str, **kwargs) -> set:
        if value in EMPTY_VALUES:
            return self.default
        if value is None:
            values = []
        elif isinstance(value, list):
            values = value
        else:
            values = value.split(self.separator)
        res = set()
        for v in values:
            if self.trim:
                v = v.strip()
            converted_v = await self.item_type.convert(v, **kwargs)
            res.add(converted_v)
        return res
```

It returns `res`which is a `set`.

Reply:
You are right. The web interface was only showing the top part of the function. So I thought lists were returned.